torchx/schedulers/ray/ray_driver.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import contextlib
+import importlib
+import json
+import logging
+import os
+import sys
+from typing import Dict, List, Optional
+
+import ray
+from ray.train.utils import get_address_and_port
+from ray.util.placement_group import PlacementGroup
+from torchx.schedulers.ray.ray_common import RayActor
+
+_logger: logging.Logger = logging.getLogger(__name__)
+_logger.setLevel(logging.INFO)
+
+
+@contextlib.contextmanager
+def redirect_argv(args: List[str]):  # pyre-ignore[3]
+    _argv = sys.argv[:]
+    sys.argv = args
+    yield
+    sys.argv = _argv
+
+
+@ray.remote
+class CommandActor:  # pragma: no cover
+    def __init__(self, command: str, env: Dict[str, str]) -> None:
+        self.args: List[str] = command.split(" ")
+        self.path: str = self.args[0]
+
+        for k, v in env.items():
+            os.environ[k] = v
+
+    def run_command(self) -> None:
+        spec: Optional[
+            importlib.machinery.ModuleSpec
+        ] = importlib.util.spec_from_file_location("__main__", self.path)
```

Comment:
curious, why are we using `_main_` here, what if we are using something else, will it not work?